cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1305,6 +1313,245 @@ def _check_column_availability(self):
                 if "add_labels" in index_def:
                     self._check_single_column_availability(component, index_def["add_labels"])
 
+    def _check_spatial(self):
+        """
+        Validate spatial-related values of the AnnData object. Validation is not defined in schema definition yaml.
+        Errors are added to self.errors.
+
+        :rtype none
+        """
+
+        # Exit if uns is not specified. Error is reported in core validate functionality.
+        uns_component = getattr_anndata(self.adata, "uns")
+        if uns_component is None:
+            return
+
+        # uns spatial validation is dependent on obs.assay_ontology_term_id; exit if not specified. Error is
+        # reported in core validate functionality.
+        obs_component = getattr_anndata(self.adata, "obs")
+        if obs_component is None or "assay_ontology_term_id" not in obs_component:
+            return
+
+        # spatial is forbidden if assay it not a supported spatial assay.
+        uns_spatial = self.adata.uns.get("spatial")
+        is_supported_spatial_assay = self._is_supported_spatial_assay()
+        if uns_spatial is not None and not is_supported_spatial_assay:
+            self.errors.append(
+                "uns['spatial'] is only allowed for obs['assay_ontology_term_id'] values "
+                "'EFO:0010961' (Visium Spatial Gene Expression) and 'EFO:0030062' (Slide-seqV2)."
+            )
+            return
+
+        # Exit if we aren't dealing with a supported spatial assay as no further checks are necessary.
+        if not is_supported_spatial_assay:
+            return
+
+        # spatial is required for supported spatial assays.
+        if uns_spatial is None:
+            self.errors.append(
+                "uns['spatial'] is required for obs['assay_ontology_term_id'] values "
+                "'EFO:0010961' (Visium Spatial Gene Expression) and 'EFO:0030062' (Slide-seqV2)."
+            )
+            return
+
+        # is_single is required.
+        if "is_single" not in uns_spatial:
```

Comment:
If I set `uns['spatial']` to an int, bool, or float, that raises a TypeError and stops validation: "TypeError: argument of type {'int' | 'bool' | 'float'} is not iterable." 

Other collections like lists, numpy arrays, and pandas dataframes are handled, validation completes, and the errors with validation are displayed correctly.

I've covered most of the testing but I'm still working on some items. I wanted to comment as I've uncovered issues. Let me know if you prefer the more formal review or different way to do this.